File: src/main/java/io/srmppn/HappyDorm/room/api/RoomCreation.java
package io.srmppn.HappyDorm.room.api;

import org.axonframework.modelling.command.TargetAggregateIdentifier;

import java.util.UUID;

public class RoomCreation {

    public static class CreateRoomCommand {
        @TargetAggregateIdentifier
        private UUID roomId;

        public CreateRoomCommand(UUID roomId) {
            this.roomId = roomId;
        }

        public UUID getRoomId() {
            return roomId;
        }

        public void setRoomId(UUID roomId) {
            this.roomId = roomId;
        }
    }

    public static class RoomCreatedEvent {
        private UUID roomId;

        public RoomCreatedEvent(UUID roomId) {
            this.roomId = roomId;
        }

        public UUID getRoomId() {
            return roomId;
        }

        public void setRoomId(UUID roomId) {
            this.roomId = roomId;
        }
    }
}

File: src/main/java/io/srmppn/happydorm/room/api/room-creation.kt
package io.srmppn.happydorm.room.api

import org.axonframework.modelling.command.TargetAggregateIdentifier
import org.axonframework.serialization.Revision
import java.util.*

class CreateRoomCommand(
  @TargetAggregateIdentifier
  val roomId: UUID
)

@Revision("1.0")
class RoomCreatedEvent(
  val roomId: UUID
)